docs/src/main/asciidoc/security-authorization.adoc

```diff
@@ -270,6 +270,8 @@ public class SubjectExposingResource {
 <4> This call to obtain the user principal will return null if the caller is unauthenticated, non-null if the caller is authenticated.
 <5> The `/subject/denied` endpoint disallows any access regardless of whether the call is authenticated by specifying the `@DenyAll` annotation.
 
+CAUTION: RBAC annotations on CDI beans are not supported on IO thread if secured method returns "something" in a synchronous manner and lazy authentication is in use, as resolving the identity may block the thread.
```

Comment:
```suggestion
CAUTION: RBAC annotations on CDI beans are not supported on IO thread if a non-void secured method returns a value synchronously and proactive authentication is disabled, as resolving the security identity may block the thread.
```

Reply:
@michalvavrik I've suggested a minor edit, as for the moment in the docs we just talk about the proactive auth being on/off, so `lazy authentication` might be an unknown concept for some users.

That said, should we have this caution added at https://quarkus.io/guides/security-built-in-authentication#proactive-authentication instead, and perhaps link to this section from here where this PR adds this caution  ?

It also feels we should show an example of the incorrectly setup CDI bean followed by  another example showing how to get it done correctly, using the blocking annotation, etc

Reply:
This is very helpful, thank you. I tried to address it, let's have a look again once you find a time, thanks.